cmd/agent/app/reporter/grpc/collector_proxy.go

```diff
@@ -51,11 +49,13 @@ func NewCollectorProxy(o *Options, logger *zap.Logger) (*ProxyBuilder, error) {
 		r.InitialAddrs(resolvedAddrs)
 		conn, _ = grpc.Dial(r.Scheme()+":///round_robin", grpc.WithInsecure(), grpc.WithBalancerName(roundrobin.Name))
 	} else {
+		// It does not return error if the collector is not running
+		// a way to fail immediately is to call WithBlock and WithTimeout
 		conn, _ = grpc.Dial(o.CollectorHostPort[0], grpc.WithInsecure())
 	}
 	return &ProxyBuilder{
-		reporter: NewReporter(conn, logger),
-		manager:  NewSamplingManager(conn)}, nil
+		reporter: aReporter.WrapWithMetrics(NewReporter(conn, logger), "grpc-reporter", mFactory),
+		manager:  httpserver.WrapWithMetrics(NewSamplingManager(conn), mFactory)}, nil
```

Comment:
* not sure why namespace is passed to `reporter.WrapWithMetrics` but not to `httpserver.WrapWithMetrics`
* also, strictly speaking you can just do `mFactory.Namespace("grpc-reporter", nil)` here and not expose WrapWithMetrics to an extra parameter.

Reply:
> not sure why namespace is passed to reporter.WrapWithMetrics but not to httpserver.WrapWithMetrics

First of all I wanted to keep the old metric names for tchannel reporter. I am wondering whether scoping them based on the propag adds any value. We can make them generic or scope them. I am not sure what is better. The propag itself could also add its specific metrics.

cc @objectiser

> also, strictly speaking you can just do mFactory.Namespace("grpc-reporter", nil) here and not expose WrapWithMetrics to an extra parameter.

Can do, passing a string namespace makes it cleaner that it has to be scoped.

Reply:
Do you have examples of the metrics for reporter and manager to look at? I assume the reporter will be `jaeger_agent_grpc_reporter...` similar to `jaeger_agent_tc_reporter...`?

I think just passing the factory i.e. `mFactory.Namespace("grpc-reporter", nil)` is better.

Reply:
The reporter from master emits this:
```
jaeger_agent_tchannel_reporter_batch_size{format="jaeger"} 0
jaeger_agent_tchannel_reporter_batch_size{format="zipkin"} 0
jaeger_agent_tchannel_reporter_batches_failures
jaeger_agent_tchannel_reporter_batches_failures{format="jaeger"} 0
jaeger_agent_tchannel_reporter_batches_failures{format="zipkin"} 0
jaeger_agent_tchannel_reporter_batches_submitted
jaeger_agent_tchannel_reporter_batches_submitted{format="jaeger"} 0
jaeger_agent_tchannel_reporter_batches_submitted{format="zipkin"} 0
jaeger_agent_tchannel_reporter_spans_failures
jaeger_agent_tchannel_reporter_spans_failures{format="jaeger"} 0
jaeger_agent_tchannel_reporter_spans_failures{format="zipkin"} 0
jaeger_agent_tchannel_reporter_spans_submitted
jaeger_agent_tchannel_reporter_spans_submitted{format="jaeger"} 0
jaeger_agent_tchannel_reporter_spans_submitted{format="zipkin"} 0
```
manager:
```
jaeger_agent_collector_proxy{endpoint="baggage",result="err"} 0
jaeger_agent_collector_proxy{endpoint="baggage",result="ok"} 0
jaeger_agent_collector_proxy{endpoint="sampling",result="err"} 0
jaeger_agent_collector_proxy{endpoint="sampling",result="ok"} 0
```

Currently the manager does not scope it based on propagation. So do we want to have
`jaeger_agent_<propag>_reporter_spans` or `jaeger_agent_reporter_spans` and analogically the same for collector_proxy/manager (might be a subject to renaming)

Reply:
I'd vote for making `<propag>` a label, so `jaeger_agent_reporter_spans`.

Reply:
Yes a different metric created. Yes to wildcard - it is achieved by "ignoring" the tag, which results in the metrics being aggregated.

Reply:
thanks then this seems like a right way to go

Reply:
changed, passing the factory is better :+1:

Reply:
instead of `propag` I have used `protocol`

Reply:
Although we use `protocol` tag to distinguish thrift binary/compat.

```
jaeger_agent_thrift_udp_server_packet_size{model="jaeger",protocol="binary"} 0
jaeger_agent_thrift_udp_server_packet_size{model="jaeger",protocol="compact"} 0
jaeger_agent_thrift_udp_server_packet_size{model="zipkin",protocol="compact"} 0
jaeger_agent_thrift_udp_t_processor_close_time_bucket{model="jaeger",protocol="binary",le="0.25"}
...
```

We should find a different tag key. Any suggestions? `rpc`, `propag`?..

Reply:
Here (UDP server) this refers to inbound protocol. In the Reporter it's outbound, so still works for me.